api/task_lifecycle_client.go

```diff
@@ -166,10 +166,10 @@ func (d *TaskLifecycleHTTPClient) Do(req *http.Request) (*http.Response, error)
 		}
 
 		if errMsg == "" {
-			return nil, fmt.Errorf("request returned %d status code", resp.StatusCode)
+			return nil, fmt.Errorf("%d %s", resp.StatusCode, http.StatusText(resp.StatusCode))
 		} else {
-			return nil, fmt.Errorf("request returned %d status code with error: %s",
-				resp.StatusCode, errMsg)
+			return nil, fmt.Errorf("%d %s: %s",
+				resp.StatusCode, http.StatusText(resp.StatusCode), errMsg)
```

Comment:
I might be wrong about this, but I think there's a field in the `resp` that has the code and the text combined.

Reply:
`resp.Status` is what I was thinking of!
https://pkg.go.dev/net/http#Response

Reply:
yep, good call! Changed it, it's `resp.Status`